internal/ethapi/api.go

```diff
@@ -1557,14 +1557,23 @@ func RPCMarshalHeader(head *types.Header) map[string]interface{} {
 	if head.RequestsHash != nil {
 		result["requestsRoot"] = head.RequestsHash
 	}
+	if isCelo1 {
+		result["totalDifficulty"] = (*hexutil.Big)(new(big.Int).Add(head.Number, common.Big1))
+	}
 	return result
 }
 
+// isCelo1Block determines whether the given block is a Celo1 block
+// based on the chain config and the provided block time
+func isCelo1Block(config *params.ChainConfig, blockTime uint64) bool {
```

Comment:
Couldn't you use `!config.IsCel2(blockTime)` instead of adding this function?

Reply:
Thee's the extra condition, that `config.Cel2Time` is set, if we just used `!config.IsCel2(blockTime)` this behaviour would be applied to all OP forks.